src/mlpack/methods/ann/layer/sequential_impl.hpp

```diff
@@ -99,12 +105,23 @@ void Sequential<InputDataType, OutputDataType, CustomLayers...>::Forward(
     }
   }
 
-if (!reset)
-{
-  reset = true;
-}
+  if (!reset)
+  {
+    reset = true;
+  }
 
   output = boost::apply_visitor(outputParameterVisitor, network.back());
+
+  if (residual)
+  {
+    if (arma::size(output) != arma::size(input))
+    {
+      Log::Fatal << "The sizes of the output and input matrices of the residual"
+          << " block should be equal. Please examine the network architecture."
+          << std::endl;
+    }
+    output += input;
```

Comment:
Should this be a `-` not `+`?  When I think residual, I think "output - input".  But I could be wrong, I am not fully familiar with how this is typically done.

Reply:
@rcurtin, I am not sure what context you are talking about. Maybe in a more general sense, residual means  "output - input", but I am pretty sure in this case it's "output + input".
@ShikharJ  Done!

Reply:
Right, sorry about my comment, I was not familiar.  It looks good to me now that I look through the paper. :+1:

Reply:
No problem. :)